plugins/modules/cloud/opennebula/one_vm.py

```diff
@@ -1282,6 +1283,29 @@ def resume_vms(module, client, vms):
     return changed
 
 
+def release_vm(module, client, vm):
+    vm = client.vm.info(vm.ID)
+    changed = False
+
+    state = vm.STATE
+    if state != VM_STATES.index('HOLD'):
+        module.fail_json(msg="Cannot perform action 'release' because VM is not in state 'HOLD'.")
```

Comment:
How about simply not doing something, instead of failing? (Just as a general comment, I lack any domain-specific knowlege about this operation, so maybe this makes no sense at all :) )

(This also makes more sense for a state `released`, as opposed for an operation `release`.)